src/tools/compiletest/src/directives/handlers.rs

```diff
@@ -67,6 +75,338 @@ fn make_directive_handlers_map() -> HashMap<&'static str, Handler> {
                 |r| r,
             );
         }),
+        handler(DOC_FLAGS, |config, ln, props| {
+            config.push_name_value_directive(ln, DOC_FLAGS, &mut props.doc_flags, |r| r);
+        }),
+        handler(COMPILE_FLAGS, |config, ln, props| {
+            if let Some(flags) = config.parse_name_value_directive(ln, COMPILE_FLAGS) {
+                let flags = split_flags(&flags);
+                for (i, flag) in flags.iter().enumerate() {
+                    if flag == "--edition" || flag.starts_with("--edition=") {
+                        panic!("you must use `//@ edition` to configure the edition");
+                    }
+                    if (flag == "-C"
+                        && flags.get(i + 1).is_some_and(|v| v.starts_with("incremental=")))
+                        || flag.starts_with("-Cincremental=")
+                    {
+                        panic!(
+                            // (preserve line breaks)
+                            "you must use `//@ incremental` to enable incremental compilation"
+                        );
+                    }
+                }
+                props.compile_flags.extend(flags);
```

Comment:
Suggestion: can you add a FIXME here to pull out compile-flag normalizations and then denylisting?